### ISSUE TITLE
Doesn't work on Alpine linux (musl)

### ISSUE THREAD

**Issue by timsofteng** [Schniz/fnm#694]
I've installed it as guide described.
```
>  curl -fsSL https://fnm.vercel.app/install | zsh
 
>  fnm install 16
 
>   ./.fnm/node-versions/v16.14.0/installation/bin/node -v
zsh: no such file or directory: ./.fnm/node-versions/v16.14.0/installation/bin/node

**Comment by everyx** [comment]
I found a repo with Alpine(musl) support patch: https://github.com/shyim/fnm-alpine , and can we get Alpine official support?

**Comment by Schniz** [comment]
we need to add another rust build for that. could be a nice contribution

**Comment by arvati** [comment]
I just update fnm alpine version on my repo: https://github.com/vanaware/fnm-alpine      
You may install like this:
```sh
curl -fsSL https://vanaware.github.io/fnm-alpine/install.sh | ash
```

No needed to change rust build, basicaly just needed to change node source files to this repo:    
"https://unofficial-builds.nodejs.org/download/release/"    
and use musl version of node.     

Known bugs:     
* not able to install from "latest" keyword you must provide version number like this:   `fnm install 20`

**Comment by Vinfall** [comment]
I'm offering a _native_ solution in #1109, just three lines of code.

And following [commands.md](https://github.com/Schniz/fnm/blob/master/docs/commands.md), add these to your shell rc:

```sh
export FNM_NODE_DIST_MIRROR=https://unofficial-builds.nodejs.org/download/release
export FNM_ARCH=x64-musl
```

Reload by `exec zsh` or whatever, done.

Now you can install like `fnm install --lts` as usual.

**Comment by NamesMT** [comment]
If anyone use `Alpine` and `zsh`, you could use my simple install script, which uses @Vinfall build under-the-hood.
It basically downloads Vinfall's build and set all the required exports to zsh: `ARCH, DIST_MIRROR, PATH, fnm env --use-on-cd`

`wget https://raw.githubusercontent.com/NamesMT/images-alpine/main/scripts/install-fnm.sh -O- | sh`

**Comment by Vinfall** [comment]
Wow, I think the artifact is already expired but it seems the expiration date is 90 days by default. Since you noticed that, I've uploaded the assets to [release](https://github.com/Vinfall/fnm/releases/tag/action-build) with hashes. Maybe you'll find it useful to verify the checksum in the script.

**Comment by Vinfall** [comment]
Just FYI: #1109 is merged so in next release you can use official build and set those shell rc I mentioned in https://github.com/Schniz/fnm/issues/694#issuecomment-2049004644 to use fnm in x64 musl-libc distros.

**Comment by tindzk** [comment]
Here is a complete example that works with Alpine 3.17 and the default ash shell:

```shell
# Resolves linking errors such as:
# Error loading shared library libstdc++.so.6: No such file or directory (needed by /root/.local/state/fnm_multishells/15_1718461045839/bin/node)
apk add libstdc++6 libstdc++

mkdir bin/
cd bin/
wget https://github.com/Schniz/fnm/releases/download/v1.37.1/fnm-linux.zip
unzip fnm-linux.zip
chmod +x fnm
cd -

export PATH=$(pwd)/bin/:$PATH

eval "$(fnm env)"
export FNM_NODE_DIST_MIRROR=https://unofficial-builds.nodejs.org/download/release
export FNM_ARCH=x64-musl

cd ci/

fnm install
fnm use

npm ci
```

**Comment by Vinfall** [comment]
Yeah, I'm a bit lazy to only get rid of the hardcoded arch check. After #1271 is merged you no longer need to set `FNM_ARCH` manually. `FNM_NODE_DIST_MIRROR` is still needed since musl builds only exist in unofficial mirror.

**Comment by Schniz** [comment]
unfortunately #1271 can't be merged without supporting the formal nodejs mirror which doesn't have musl builds. we have to have multiple platform support or figure out the lack of glibc in runtime.
